modules/server/src/mapping/types.ts

```diff
@@ -0,0 +1,55 @@
+import { Client } from '@elastic/elasticsearch';
+
+export type Bucket = {
+	doc_count: number;
+	key: string;
+	belowThreshold: boolean;
+};
+
+export type Aggregation = {
+	bucket_count: number;
+	buckets: Bucket[];
+};
+
+export type Root = Record<string, unknown>;
+
+enum Missing {
+	first,
+	last,
+}
```

Comment:
common file for TS for now, there will be cleanup needed

Reply:
Are these new enums or moved from somewhere else? I would prefer us not to use TS enums, and if we do we should have constant values (ie. `first = 'first'` or `first = 0`) to prevent their literal values changing by re-ordering the enum list in the (likely to cause no error, but can cause real tricky bugs especially if enums are used between client and server). Generally speaking, prefer not to use TS enums at all. Would you consider something like:

```ts
const Modes = {
  avg: 'avg',
  max: 'max',
  min: 'min',
  sum: 'sum',
} as const satisfies Record<string, string>; // as const makes the object store literal types, not string type. satisfies provides type system checking to ensure all values are strings.
type Mode = typeof Modes[keyof typeof Modes]; // gets values as union type.
```

Reply:
Generally speaking, I don't create a dedicated "types" files, in favour of keeping types near where they need to be declared. Treat them more akin to exported functions. Theres several different concepts stored in this file, including enums used for config/input to queries, the data structure returned by our es queries, whatever Root and Context is.

I think my advice here is similar to not having "utils" files, instead name the file based on its content and export similar ideas. At the moment, this file is not too large to manage, but I always fear these become dumping grounds of any "type" information.